src/main/kotlin/io/slama/commands/rockpaperscissors_command.kt

```diff
@@ -0,0 +1,259 @@
+        if (winner != null) roundSummaries.add(previousRoundSummary)
+
+        if (winner != null && winner.score > rounds / 2) {
+            end(winner)
+            return
+        }
+
+        if (currentRound == rounds) {
+            if (player1.score > player2.score) {
+                end(player1)
+            } else if (player2.score > player1.score) {
+                end(player2)
+            } else {
+                end(null)
+            }
+            return
+        }
+
+        if (winner != null) currentRound++
+
+        player1.move = null
+        player2.move = null
+
+        event.hook.editOriginalEmbeds(
+            EmbedBuilder()
+                .setTitle(GAME_NAME)
+                .setDescription(
+                    "${if (winner != null) "$winner" else "Personne ne"} remporte le tour !\n\n" +
+                        "$previousRoundSummary\n\n" +
+                        "**Vous pouvez rejouer !**"
+                )
+                .setFooter("Partie #${gameHash()} • Round $currentRound/$rounds • En cours")
+                .setColor(EmbedColors.VIOLET)
+                .build()
+        ).queue()
+    }
+
+    private fun end(winner: RPSPlayer?) {
+        logger.info("[GAME-${gameHash()}] (Round $currentRound/$rounds) Game ended, winner: ${winner?.id}")
+
+        cancellationTask?.cancel(true)
+        event.jda.removeEventListener(this)
+        event.hook.editOriginalEmbeds(
+            EmbedBuilder()
+                .setTitle(GAME_NAME)
+                .setDescription("${if (winner != null) "$winner" else "Personne ne"} remporte la partie !\n\n")
+                .addField(
+                    "Résumé des tours",
+                    roundSummaries.joinToString("\n") { it.toString() },
+                    false
+                )
+                .addField(
+                    "Score final",
+                    "$player1 ${player1.score} - ${player2.score} $player2",
+                    false
+                )
+                .setFooter("Partie #${gameHash()} • Round $currentRound/$rounds • Terminée")
+                .setColor(if (winner != null) EmbedColors.GREEN else EmbedColors.ORANGE)
+                .build()
+        ).setActionRows().queue()
+    }
+
+    override fun onButtonClick(event: ButtonClickEvent) {
+        if (event.user.idLong != player1.id && event.user.idLong != player2.id) {
+            event.replyError("Vous n'êtes pas un joueur de cette partie !")
+                .setEphemeral(true)
+                .queue()
+            return
+        }
+
+        when (event.componentId) {
+            "rps.$gameId-rock" -> play(event, RPSMove.ROCK)
+            "rps.$gameId-paper" -> play(event, RPSMove.PAPER)
+            "rps.$gameId-scissors" -> play(event, RPSMove.SCISSORS)
+        }
+    }
+
+    private fun gameHash() = Integer.toHexString(Objects.hash(gameId, player1, player2))
```

Comment:
fine engough ; curious though : why not UUID ?

Reply:
I first though UUID was too long, and I would've need to chop down to keep just a part of the ID, so that the displayed ID is not too long. Afterward, I realised UUIDs could do the trick, assuming their high randomness.